### DISCUSSION TITLE
在技术上来说,使用不知名私人梯子和飞机场 的基于hysteria协议的梯子有数据泄露风险吗?

### DISCUSSION THREAD

**Discussion by q2333gh** [apernet/hysteria#908] [General]
好奇

**Comment by tobyxdd** [comment]
使用任何协议的任何梯子都有

**Comment by nxtreaming** [comment]
墙识别翻墙行为不是通过解密数据，而是通过流量特征。

**Reply by q2333gh** [reply]
我意思是, hysteria 搭建服务器. 服务器能否明文获取用户的数据.

**Reply by yh317** [reply]
现在互联网基本都是https了，顶多能看见你访问的域名。具体内容根本看不见。
除非他有外星科技把AES加密破了。

**Reply by q2333gh** [reply]
翻墙行为确实是你说得那样.完全同意.
我侧重点是服务器端偷偷获取用户的https content内容. 最近了解了一下https和各种协议的伪装方式.
发现如果没有证书是各种机场的服务器端是难以解密内容的.

**Comment by q2333gh** [comment]
自己也基于x-ui自建过机场. 就是没有真正深入代码查看是否有可能收集用户明文数据

**Comment by listenecho** [comment]
能看到你的请求日志，至于把你的请求内容解析出来？目前还不知道能不能行，但我觉得肯定行~

**Comment by q2333gh** [comment]
理论上如果是https且保证了证书安全，hy2的服务端只能获取到请求的header信息，content仍然是加密的

**Comment by xiaorong61** [comment]
可以通过 sni 发现你访问的网站，但不能破解访问的内容。
